bin/propolis-server/src/lib/initializer.rs

```diff
@@ -28,12 +28,18 @@ use propolis::hw::ibmpc;
 use propolis::hw::pci;
 use propolis::hw::ps2::ctrl::PS2Ctrl;
 use propolis::hw::qemu::pvpanic::QemuPvpanic;
-use propolis::hw::qemu::{debug::QemuDebugPort, fwcfg, ramfb};
+use propolis::hw::qemu::{
+    debug::QemuDebugPort,
+    fwcfg::{self, Entry},
+    ramfb,
+};
 use propolis::hw::uart::LpcUart;
 use propolis::hw::{nvme, virtio};
 use propolis::intr_pins;
 use propolis::vmm::{self, Builder, Machine};
-use propolis_api_types::instance_spec::{self, v0::InstanceSpecV0};
+use propolis_api_types::instance_spec::{
+    self, v0::BootDeclaration, v0::InstanceSpecV0,
```

Comment:
oh yes, one thing i'd not understood in our OpenAPI documents: why are some things named `v0::ThingV0` and some are... not? e.g. `v0::SerialPort` vs `v0::InstanceSpecV0`. would the new type be `v0::BootDeclarationV0`?

Reply:
My (possibly misguided) intention here was to separate the component definitions themselves (things like `SerialPort`) from the types that collect a bunch of components into a complete instance spec, so that defining a new "format" for a component collection wouldn't entail copying a bunch of component definitions into a new `v1` module (or having the `v1` module depend on component definitions from the `v0` module, which seemed confusing to me). 

At the risk of lulling you to sleep with more of my technical writing, there's a [theory statement](https://github.com/oxidecomputer/propolis/blob/5267be82e10d851a64196a8148893691b0b9f8ad/crates/propolis-api-types/src/instance_spec/mod.rs) in the instance spec portion of the types crate that might provide some additional color here.

The version-numbered types have the `V0` suffix because all of the types that contribute to a generated OpenAPI spec need to have distinct *names*, not just distinct Rust module paths; if you do try to refer to multiple identically-named types, Dropshot will pick one to include and discard the rest. (This is dropshot#383).

In this case, I think the boot order declaration is functionally a "component", so I don't think I would give it a suffix; it should live under the `components` part of the module tree and be referenced from within the spec structure from there. (Eventually I expect to collapse all the various component containers in `InstanceSpecV0` into a single bag of components so that it's clearer where to put new things.)